File: 1988-minimize-maximum-pair-sum-in-array/minimize-maximum-pair-sum-in-array.java
class Solution {
    public int minPairSum(int[] nums) {
        Arrays.sort(nums);
        int i = 0;
        int res = 0;
        int j = nums.length -1 ;
        while(i < j){
            res = Math.max(nums[i]+nums[j],res);
            ++i;
            --j;
        }
        return res;
    }
}

File: 1988-minimize-maximum-pair-sum-in-array/minimize-maximum-pair-sum-in-array.cpp
class Solution {
public:
    int minPairSum(vector<int>& nums) {
        sort(nums.begin(),nums.end());
        int i =0 ; 
        int j = nums.size()-1;
        int res = 0;
        while(i < j){
            res = max(nums[i]+nums[j],res);
            ++i;
            --j;
        }
        return res;
    }
};